### PR TITLE
Documentation model evaluation with example - New version

### PR DESCRIPTION
**Pull request by maylistran01**
New PR for the model evaluation documentation 

## TO-DO: 

- [ ] Convergence Diagnosis but @KaisaridiSofia is taking care of that in a PR later.
- [ ] Example for survival metrics -> need for creation of survival data in order to generate survival examples (@lea-aguilhon)

Old PR https://github.com/aramis-lab/leaspy/pull/367 was too old -> too many conflicts. That's why we created this one.

### CODE REVIEW

File: docs/models_evaluation.md [hunk 1]
```diff
@@ -1,9 +1,222 @@
...
+
+#### Mean Square Error (MSE)
+
+$$
+MSE = \frac{1}{n} \sum_{i=1}^{n} (y_i - \hat{y}_i)^2
+$$
+
+MSE penalizes larger errors more heavily than MAE, making it more sensitive to outliers {cite}`willmott2005advantages`{cite}`chai2014root`.
+
+Package: [`sklearn.metrics`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_squared_error.html)
```
Comment (caglayantuna) [inline docs/models_evaluation.md:146]: same

File: docs/models_evaluation.md [hunk 2]
```diff
@@ -1,9 +1,222 @@
...
+mse = mean_squared_error(alzheimer_df_to_pred["MMSE"], df_pred["MMSE_model1"])
+print(f"Mean Squared Error: {mse}")
+
+Mean Squared Error: 0.0041833904503857655
+```
+
+#### Residual Q-Q Plot
+A graphical tool to assess whether residuals follow a normal distribution, which is an assumption in many mixed-effects models {cite}`fox2015applied`.
+
+Package: [`statsmodels.api`](https://www.statsmodels.org/stable/generated/statsmodels.graphics.gofplots.qqplot.html)
```
Comment (caglayantuna) [inline docs/models_evaluation.md:166]: same

File: docs/models_evaluation.md [hunk 3]
```diff
@@ -1,9 +1,222 @@
...
+#### Coefficient of Determination (R²)
+R² indicates how well the model explains the variance in the observed data. Higher values (closer to 1) suggest better performance.
+
+$$
+R^2 = 1 - \frac{\sum_{i=1}^{n} (y_i - \hat{y}_i)^2}{\sum_{i=1}^{n} (y_i - \bar{y})^2}
+$$
+
+In mixed-effects models, multiple R² variants exist (e.g., marginal vs. conditional R²) to account for fixed and random effects {cite}`nakagawa2012method`.
+
+Package: [`sklearn.metrics`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html)
```
Comment (caglayantuna) [inline docs/models_evaluation.md:190]: same

File: docs/models_evaluation.md [hunk 4]
```diff
@@ -1,9 +1,222 @@
...
+
+### Events
+
+For the joint model, it is interesting to have prediction metrics on survival as well:
+
+#### Integrated Brier Score (IBS)
+The Integrated Brier Score (IBS) is a robust metric used to evaluate the predictive accuracy of survival models. 
+In survival analysis, IBS quantifies how well a model predicts the timing of events by integrating the Brier Score, a measure of prediction error for probabilistic outcomes, across all observed time points. It compares the model’s predicted survival probabilities against actual observed outcomes, penalizing discrepancies between predictions and reality. IBS accounts for censored data, ensuring reliability in real-world datasets where not all events are fully observed. IBS describes how closely predictions match reality. A low IBS indicates superior predictive performance, as it reflects smaller cumulative errors over time {cite}`graf1999assessment`.
+
+Package: [`scikit-survival`](https://scikit-survival.readthedocs.io/en/stable/api/generated/sksurv.metrics.integrated_brier_score.html)
```
Comment (caglayantuna) [inline docs/models_evaluation.md:208]: same

File: docs/models_evaluation.md [hunk 5]
```diff
@@ -1,9 +1,222 @@
...
+from sklearn.metrics import r2_score
+r2 = r2_score(alzheimer_df_to_pred["MMSE"], df_pred["MMSE_model1"])
+print(f"R² Score: {r2}")
+
+R² Score: 0.9442675319358963
+```
+
+### Events
+
+For the joint model, it is interesting to have prediction metrics on survival as well:
```
Comment (caglayantuna) [inline docs/models_evaluation.md:202]:
```suggestion
For the joint model, it is useful to have prediction metrics on survival as well:
```

**Review by caglayantuna — COMMENTED** [review]
Thanks @maylistran01 for this huge work. I made a review with suggestions. Generally, they are repeated comments.

File: docs/models_evaluation.md [hunk 6]
```diff
@@ -1,9 +1,222 @@
...
+### Frequentist Approach
+
+#### AIC
+AIC (Akaike Information Criterion) is a robust metric for model selection. It integrates the goodness-of-fit and the complexity of the model (number of features and number of patients). It has a penalty term for the number of parameters in the model, thus penalizing more complex models with unnecessary features. Lower AIC values indicate a better model {cite}`akaike1974new`.
+
+$$
+\text{AIC} = 2 \cdot (\text{nb}_{\text{features}}) - 2 \cdot \log(\text{likelihood})
+$$
+
+Package: `leaspy`
```
[the remark this answers is not in the document]
Comment (maylistran01) [reply docs/models_evaluation.md:40]: For each metrics, we mention the package we are using (sklearn.metrics, or statsmodels.api). Here for the AIC, BIC and WAIC we just need to use Leaspy.

File: docs/models_evaluation.md [hunk 7]
```diff
@@ -1,9 +1,222 @@
...
+
+$$
+\text{AIC} = 2 \cdot (\text{nb}_{\text{features}}) - 2 \cdot \log(\text{likelihood})
+$$
+
+Package: `leaspy` 
+
+```python
+# Get the negative log-likelihood
+neg_LL = model.state['nll_attach']
```
[the remark this answers is not in the document]
Comment (maylistran01) [reply docs/models_evaluation.md:44]: Yes you're right, I will do the changes.

File: docs/models_evaluation.md [hunk 8]
```diff
@@ -1,9 +1,222 @@
...
+
+This section describes the main metrics used to evaluate the quality of predictions from the Leaspy non-linear mixed-effects model.
+
+#### Mean Absolute Error (MAE)
+
+$$
+MAE = \frac{1}{n} \sum_{i=1}^{n} |y_i - \hat{y}_i|
+$$
+
+MAE is more robust to outliers and provides a straightforward sense of typical prediction error.
```
[the remark this answers is not in the document]
Comment (maylistran01) [reply docs/models_evaluation.md:119]: More robust than MSE. I will add that.

File: docs/models_evaluation.md [hunk 9]
```diff
@@ -1,9 +1,222 @@
...
+
+Package: [`sklearn.metrics`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_absolute_error.html)
```
[the remark this answers is not in the document]
Comment (maylistran01) [reply docs/models_evaluation.md:121]: Ok now I see your point. I can do:  
- Remove all the "Package: leaspy" -> because not really relevant. 
- Change all the "Package: (other package than Leaspy)" to "For more information, please see ...".
Do you agree with this?

**Review by maylistran01 — COMMENTED** [review]
Thank you for your comments ! Here are some answers.

File: docs/models_evaluation.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (caglayantuna) [reply docs/models_evaluation.md]: yes, I agree, thanks

**Comment by caglayantuna** [comment]
Thanks @maylistran01 for the answers and modifications. It is ready to merge. Should we wait for TODO list above before merging it?

**Comment by maylistran01** [comment]
@caglayantuna I think we can merge. 
I put the TODO in our meeting notes.
